tiledb/api/src/range.rs

```diff
@@ -57,10 +57,42 @@ pub enum SingleValueRange {
 }
 
 impl SingleValueRange {
+    /// Returns the number of cells spanned by this range if it is an integral range
+    pub fn num_cells(&self) -> Option<u128> {
+        let (low, high) = crate::single_value_range_go!(self, _DT : Integral, start, end,
+            (i128::from(*start), i128::from(*end)),
+            return None
+        );
+        Some(1 + (high - low) as u128)
+    }
+
+    pub fn is_integral(&self) -> bool {
+        matches!(
+            self,
+            Self::UInt8(_, _)
+                | Self::UInt16(_, _)
+                | Self::UInt32(_, _)
+                | Self::UInt64(_, _)
+                | Self::Int8(_, _)
+                | Self::Int16(_, _)
+                | Self::Int32(_, _)
+                | Self::Int64(_, _)
+        )
+    }
+
     pub fn check_datatype(&self, datatype: Datatype) -> TileDBResult<()> {
         check_datatype!(self, datatype);
         Ok(())
     }
+
+    /// Merges this range into the `other` list of sorted, non-overlapping ranges and returns the result.
```

Comment:
I added this stub while thinking about "do multiple writes in sequence and read back to make everything checked out" but decided to restrict the scope of this PR.  Will remote it.